src/Components/Server/src/Circuits/DefaultCircuitFactory.cs

```diff
@@ -44,13 +45,15 @@ public override CircuitHost CreateCircuitHost(HttpContext httpContext, IClientPr
             }
 
             var scope = _scopeFactory.CreateScope();
-            var jsRuntime = new RemoteJSRuntime(client);
+            var environment = scope.ServiceProvider.GetRequiredService<ComponentEnvironment>();
+            InitializeEnvironment(environment, client);
+
             var rendererRegistry = new RendererRegistry();
             var dispatcher = Renderer.CreateDefaultDispatcher();
             var renderer = new RemoteRenderer(
                 scope.ServiceProvider,
                 rendererRegistry,
-                jsRuntime,
+                environment.JSRuntime,
```

Comment:
Yeah this is not right. The circuit host is the owner of this state.

Reply:
The circuit is the one that sets up the environment a few lines above. It's still the owner.